src/res_parsetree_viewer.ml

```diff
@@ -200,6 +200,24 @@ let isHuggablePattern pattern =
   | Ppat_construct _ -> true
   | _ -> false
 
+let isExponentiationLikeOperator operator =
+   String.length operator >= 2 &&
+   String.unsafe_get operator 0 = '*' &&
+   String.unsafe_get operator 1 = '*'
+
+(* TODO: unfinished paper lexing *)
+let isAdditionLikeOperator operator =
+  String.length operator > 0 && (
+    let firstChar = String.unsafe_get operator 0 in
+    firstChar == '+' || firstChar == '-'
+  )
+
+let isMultiplicationLikeOperator operator =
+  String.length operator > 0 && (
+    let firstChar = String.unsafe_get operator 0 in
+    firstChar == '*' || firstChar == '/' || firstChar == '%'
```

Comment:
Nit: operators such as `**` are both exponentiation and multiplication at the moment.